sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -119,4 +123,195 @@ object IntervalUtils {
       case _: IllegalArgumentException => null
     }
   }
+
+  private def toLongWithRange(
+      fieldName: String,
+      s: String,
+      minValue: Long,
+      maxValue: Long): Long = {
+    val result = if (s == null) 0L else s.toLong
+    require(minValue <= result && result <= maxValue,
+      s"$fieldName $result outside range [$minValue, $maxValue]")
+
+    result
+  }
+
+  private val yearMonthPattern = "^\\s*([+|-])?(\\d+)-(\\d+)\\s*$".r
```

Comment:
The previous regex does not have the `\\s+` at the beginning and end. This is because we get the string from parser which already trims the white spaces. Shall we change it back?

Reply:
Previously, we trim the input explicitly: https://github.com/apache/spark/pull/26261/commits/b48236787b7d4c514d09ab1f6a4264d3f7b27048 . I just trim by the regexp.

Reply:
I will revert the regexp and add the assert: `assert(input.length == input.trim.length)`

Reply:
I will do the same in `fromDayTimeString()`